webapp/channels/src/components/admin_console/ip_filtering/index.tsx

```diff
@@ -0,0 +1,275 @@
+    const [saveNeeded, setSaveNeeded] = useState(false);
+    const [currentUsersIP, setCurrentUsersIP] = useState<string | null>(null);
+    const [saving, setSaving] = useState<boolean>(false);
+    const [filterToggle, setFilterToggle] = useState<boolean>(false);
+
+    useEffect(() => {
+        getIPFilters((data: AllowedIPRange[]) => {
+            setIpFilters(data);
+            setOriginalIpFilters(data);
+        });
+
+        getCurrentIP((res: FetchIPResponse) => {
+            setCurrentUsersIP(res.ip);
+        });
+    }, []);
+
+    useEffect(() => {
+        if (ipFilters === null || originalIpFilters === null) {
+            return;
+        }
+
+        // Check if the ipFilters list differs from the originalIpFilters list
+        const haveFiltersChanged = JSON.stringify(ipFilters) !== JSON.stringify(originalIpFilters);
+        setSaveNeeded(haveFiltersChanged);
+    }, [ipFilters, originalIpFilters]);
+
+    const currentIPIsInRange = () => {
+        if (!filterToggle) {
+            return true;
+        }
+        if (!ipFilters?.length) {
+            return true;
+        }
+        return ipFilters !== null && currentUsersIP !== null && isIPAddressInRanges(currentUsersIP, ipFilters);
+    };
+
+    useEffect(() => {
+        if (!ipFilters?.length) {
+            return;
+        }
+        setFilterToggle(ipFilters?.some((filter: AllowedIPRange) => filter.enabled === true) ?? false);
+    }, [ipFilters]);
+
+    useEffect(() => {
+        if (filterToggle === false) {
+            setIpFilters(ipFilters?.map((filter: AllowedIPRange): AllowedIPRange => {
+                return {
+                    ...filter,
+                    enabled: false,
+                };
+            }) || []);
+        } else {
+            setIpFilters(ipFilters?.map((filter: AllowedIPRange): AllowedIPRange => {
+                return {
+                    ...filter,
+                    enabled: true,
+                };
+            }) || []);
+        }
+    }, [filterToggle]);
+
+    function handleEditFilter(filter: AllowedIPRange, existingRange?: AllowedIPRange) {
+        setIpFilters((prevIpFilters) => {
+            if (!prevIpFilters) {
+                return [filter];
+            }
+            const index = prevIpFilters.findIndex((f) => f.cidr_block === existingRange?.cidr_block);
+            if (index === -1) {
+                return null;
+            }
+            const updatedFilters = [...prevIpFilters];
+            updatedFilters[index] = filter;
+            return updatedFilters;
+        });
+        setSaveNeeded(true);
+    }
+
+    function showAddModal() {
+        dispatch(openModal({
+            modalId: ModalIdentifiers.IP_FILTERING_ADD_EDIT_MODAL,
+            dialogType: IPFilteringAddOrEditModal,
+            dialogProps: {
+                currentIP: currentUsersIP!,
+                onClose: () => dispatch(closeModal(ModalIdentifiers.IP_FILTERING_ADD_EDIT_MODAL)),
```

Comment:
The `ModalController` automatically passes an `onExited` prop to each modal that it mounts, which dispatches `closeModal` with the `modalId` passed from here. If you use that, you don't actually don't need any of these `onClose` props

Reply:
Nice! Done.